landlab/components/species_evolution/zone_controller.py

```diff
@@ -0,0 +1,416 @@
+    example, the function returns True where node values are greater than 0.
+    Nodes marked with an ``*`` are nodes where a zone exists and the mask is
+    True. All other nodes are marked with a ``·``. A zone is created for each
+    cluster of continuous nodes where the mask is True.
+
+    values         mask returned
+    evaluated      by zone function
+    0 0 0 0 0 0    · · · · · ·
+    0 0 0 5 4 2    · · · * * *
+    0 6 0 4 6 0    · * · * * ·
+    0 2 0 0 0 0    · * · · · ·
+    0 0 4 0 4 0    · · * · * ·
+    0 0 0 0 0 0    · · · · · ·
+
+    The above example is continued in the following four diagrams that
+    demonstrate how individual zones are identified. Each zone is marked with a
+    ``x``, ``o``, ``+``, or ``@``. Clusters can be identified using ``D8``
+    where diagonal neighbors are included or ``D4`` where diagonal neighbors
+    are excluded. A minimum zone area can be enforced with the ``minimum_area``
+    parameter.
+
+    D8             D4             D8             D4
+    min area = 0   min area = 0   min area = 2   min area = 2
+    · · · · · ·    · · · · · ·    · · · · · ·    · · · · · ·
+    · · · + + +    · · · + + +    · · · + + +    · · · + + +
+    · x · + + ·    · x · + + ·    · x · + + ·    · x · + + ·
+    · x · · · ·    · x · · · ·    · x · · · ·    · x · · · ·
+    · · x · o ·    · · @ · o ·    · · x · · ·    · · · · · ·
+    · · · · · ·    · · · · · ·    · · · · · ·    · · · · · ·
+
+    The grid boundary affects zone creation. Zones can include nodes along the
+    boundary, although because boundary nodes are not associated with cells
+    that have area, boundary nodes do not contribute to the area summation of
+    clusters. The area summation only takes into account the cells associated
+    with core nodes.
+
+    Creation of zones along boundaries is illustrated below. A zone extent mask
+    different from the above example was produced by the hypothetical zone
+    function here. Again ``*`` indicates where a zone can exist. Distinct zones
+    that are created are in the other four grid schematics, where the symbols,
+    ``$`` and ``#`` indicate zones in addition to the symbols defined above.
+    Individual zone masks and the count of zones are affected by the use of
+    ``D8`` or ``D4`` along with the minimum area parameter, especially when
+    zone clusters are along grid boundaries.
+
+    zone function  D8             D4             D8             D4
+    returned mask  min area = 0   min area = 0   min_area = 2   min_area = 2
+    * · · * * ·    + · · x x ·    + · · x x ·    + · · · · ·    · · · · · ·
+    · * · · * ·    · + · · x ·    · # · · x ·    · + · · · ·    · # · · · ·
+    · * · · · ·    · + · · · ·    · # · · · ·    · + · · · ·    · # · · · ·
+    * · · · * *    + · · · o o    $ · · · o o    + · · · o o    · · · · o o
+    · · · · * ·    · · · · o ·    · · · · o ·    · · · · o ·    · · · · o ·
+    · * * · · ·    · @ @ · · ·    · @ @ · · ·    · · · · · ·    · · · · · ·
```

Comment:
In the D8 min_area = 2 case I find it a bit confusing that you get rid of the "three x" case that exists in the D8, min_area = 0 case. I think you do this because two of the x's don't count because they are on the perimeter, but then do the pluses that are on the perimeter really count?

Reply:
Thanks for the helpful questions.

The three x's are not removed in the D8, min_area = 0. They are removed in the D8, min_area = 2 case because only one of the x's are not on the perimeter.

In which case are you asking about the +'s? They are only removed in the D4, min_area = 2 case because the single +'s are diagonal from other +'s, therefore are in clusters below the area threshold.